cmd/api/src/analysis/ad/adcs_integration_test.go

```diff
@@ -896,6 +896,235 @@ func TestADCSESC4(t *testing.T) {
 				require.True(t, results.Contains(harness.ESC4Template4.Group13))
 			}
 
+			return nil
+		})
+
+	})
+
+}
+
+func TestADCSESC4Composition(t *testing.T) {
```

Comment:
Since we are setting up the same harnesses as the ESC4 post processing tests, could we instead move the test blocks that are in this function after the post processing assertions in the other? For example moving lines 941-967 to line 730. This could reduce testing overhead but wanted to ask in case there was some drawback to this.

Reply:
the reason i didn't do this is because the tests for PostADCSESC4() and for ESC4Composition() are several hundred lines, and even though they share some harness setup, this felt like the most natural seam between them.

btw all 5 of the composition tests use `ESC4Template1` harness, so theres only a one time setup cost/boilerplate. 
all of the composition tests use the template1 harness because template1 captures all of the different scenarios that esc4 manifests.

Reply:
Yea it does make them more distinct and I had not noticed that they were all template1 in the composition tests. We aren't running into testing resource limits so this sounds good to me. Thank for the breakdown